Mathlib/Tactic/Rewrites.lean

```diff
@@ -134,27 +135,25 @@ This core function returns a monadic list, to allow the caller to decide how lon
 See also `rewrites` for a more convenient interface.
 -/
 def rewritesCore (lemmas : DiscrTree (Name × Bool × Nat) s × DiscrTree (Name × Bool × Nat) s)
-    (goal : MVarId) (target : Expr) : ListM MetaM RewriteResult := ListM.squash do
-
+    (ctx : MetavarContext) (goal : MVarId) (target : Expr) :
```

Comment:
Just to check, the point of having `rewritesCore` take a `MetavarContext` is that `ListM` does not checkpoint it, so you can't have the first line in the `do` be `let ctx <- getMCtx`, right?

Reply:
That's right. I should add a comment explaining this carefully, as it confused me thoroughly. :-)